worktree.go

```diff
@@ -0,0 +1,294 @@
+package git
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+	"time"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/index"
+	"gopkg.in/src-d/go-git.v4/plumbing/object"
+
+	"srcd.works/go-billy.v1"
+)
+
+var ErrWorktreeNotClean = errors.New("worktree is not clean")
+
+type Worktree struct {
```

Comment:
the custom Worktree, will be advanced feature to be develop in the future, since this require custom `.git` folders